pkg/sbom/io/encode.go

```diff
@@ -86,7 +86,10 @@ func (e *Encoder) rootComponent(r types.Report) (*core.Component, error) {
 	case artifact.TypeCycloneDX, artifact.TypeSPDX:
 		// When we scan SBOM file
 		if r.BOM != nil {
-			return r.BOM.Root(), nil
+			// If SBOM file doesn't contain root component - use filesystem
+			if bomRoot := r.BOM.Root(); bomRoot != nil {
```

Comment:
```suggestion
			if r.BOM != nil && rBOM.Root() != nil {
				return r.BOM.Root(), nil
```

Reply:
updated in https://github.com/aquasecurity/trivy/pull/7051/commits/c4c413abd190140aa0453fd13f5d7e79b883f050